tests/integration/suite/test_node.py

```diff
@@ -160,13 +160,21 @@ def _node_template():
     node_template = wait_for(_node_template,
                              fail_handler=lambda:
                              'failed to create node template')
-    wait_remove_resource(node_template)
 
     node_pool = client.create_node_pool(
         nodeTemplateId=node_template.id,
         hostnamePrefix="test1",
         clusterId="local")
+
+    def node_available():
+        if len(client.list_node(nodePoolId=node_pool.id).data):
+            return client.list_node(nodePoolId=node_pool.id).data[0]
```

Comment:
If ```list_node``` makes an API call rather than grabs an object locally, I'd cache it a line above...

```python
node = client.list_node(nodePoolId=node_pool.id)
if len(node.data):
    return node.data[0]
```